docs/guides/go/realtime-messaging.mdx

```diff
@@ -0,0 +1,298 @@
+    err := connections.Set(context.TODO(), ctx.Request.ConnectionID(), map[string]interface{}{
+      "connectionId": ctx.Request.ConnectionID(),
+    })
+    if err != nil {
+      return
+    }
+  })
+
+  ws.On(websockets.EventType_Disconnect, func(ctx *websockets.Ctx) {
+    err := connections.Delete(context.TODO(), ctx.Request.ConnectionID())
+    if err != nil {
+      return
+    }
+  })
+
+  ws.On(websockets.EventType_Message, func(ctx *websockets.Ctx) {
+    connectionStream, err := connections.Keys(context.TODO())
+    if err != nil {
+      return
+    }
+
+    senderId := ctx.Request.ConnectionID()
+
+    for {
+      connectionId, err := connectionStream.Recv()
+      if err != nil {
+        break
+      }
+
+      if connectionId == senderId {
+        continue
+      }
+
+      message := fmt.Sprintf("%s: %s", senderId, ctx.Request.Message())
+      err = ws.Send(context.TODO(), connectionId, []byte(message))
+      if err != nil {
+        return
+      }
+    }
+  })
+
+  nitric.Run()
+}
+```
+
+</details>
+
+Do a quick `go mod tidy` to make sure all new dependencies are resolved.
+
+## Ok, let's run this thing!
+
+Now that you have your WebSocket application defined with handlers for each event, it's time to test it locally. Update your `nitric.yaml` to point to the service at `websockets/main.go`:
+
+```yaml
+services:
+  - match: websockets/*
+    runtime: go
+    start: go run ./$SERVICE_PATH
+```
+
+```bash
+nitric start
+```
+
+Once it starts, the application will be ready to accept WebSocket connections. You can use a WebSocket client like Postman or any other WebSocket tool to test the application. Alternatively, you can use the [Nitric Dashboard](/get-started/foundations/projects/local-development#local-dashboard).
+
+We will keep it running for our tests.
+
+## Deploy to the cloud
+
+At this point, you can deploy what you've built to any of the supported cloud providers. To do this, start by setting up your credentials and configuration for [AWS](/providers/pulumi/aws).
+
+Next, we'll need to create a `stack`. A stack represents a deployed instance of an application, which is a key value store of resources defined in your project. You might want separate stacks for each environment, such as stacks for `dev`, `test`, and `prod`. For now, let's start by creating a `dev` stack.
+
+The `stack new` command below will create a stack named `dev` that uses the `aws` provider.
+
+```bash
+nitric stack new dev aws
+```
+
+Continue by checking your stack file `nitric.dev.yaml` and adding in your preferred region. Let's use `us-east-1`.
+
+### AWS
+
+<Note>
+  You are responsible for staying within the limits of the free tier or any
+  costs associated with deployment.
+</Note>
+
+We called our stack `dev`. Let's try deploying it with the `up` command:
+
+```bash
+nitric up
+```
+
+When the deployment is complete, go to the relevant cloud console and you'll be able to see and interact with your WebSocket application.
```

Comment:
Maybe we should include a screenshot or mp4 screen recording of this step to help navigate the AWS console?

Alternatively we should create a docs page showing how to find the resource group nitric will deploy - that's pretty useful for finding all deployed resources in a stack and could be shared across multiple guides since it's consistent. I'll create a new issue for that if you agree (or you can).

Reply:
https://github.com/nitrictech/docs/issues/642